src/parser/priest/discipline/modules/spells/Penance.tsx

```diff
@@ -41,10 +39,11 @@ class Penance extends Analyzer {
   }
 
   get currentBoltNumber() {
+    // @ts-ignore
     return [...this.eventGrouper].slice(-1)[0].length - 1; // -1 here for legacy code
   }
 
-  on_byPlayer_damage(event) {
+  on_byPlayer_damage(event: PenanceDamageEvent) {
```

Comment:
`DamageEvent`

Reply:
I can't use the typechecking to guard this function, this function is what adds the field to the event.